### ISSUE TITLE
[HOLD for payment 2024-02-14] [$500] Expense - "Enter a merchant" is still shown when the expense is paid

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#34415]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.24-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:** 

## Action Performed:
1. [Member] Go to workspace chat and create a scan request.
2. [Member] Manually enter the amount of the scanning request, while leaving Merchant empty.
3. [Admin] Pay the request.
4. [Admin] Navigate to request details view.

## EXPECTED RESULT
The Merchant row will not show "Enter a merchant" since admin can no longer edit the paid expense.

## ACTUAL RESULT
The Merchant row shows "Enter a merchant" with a red dot when there is no way to edit merchant for a paid expense

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/a424c4d1-0c15-45d7-9700-bff6c6bf5823

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018efb2cb983d9645c</li>
        <li>Upwork Job ID: 1749775068612321280</li>
        <li>Last Price Increase: 2024-01-23</li>
<li>Automatic offers: </li>
<ul>
<li>c3024 | Reviewer | 28128948</li>
<li>DylanDylann | Contributor | 28128950</li>
</ul></li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?
https://github.com/Expensify/App/blob/b6a18b185f485cf17474c18add56359f4624b56a/src/components/ReportActionItem/MoneyRequestView.js#L317
In here we don't check if the request money is paid or not
### What changes do you think we should make in order to solve the problem?

We should add logic to check if the request money is paid or not in here
https://github.com/Expensify/App/blob/b6a18b185f485cf17474c18add56359f4624b56a/src/components/ReportActionItem/MoneyRequestView.js#L317
### What alternative solutions did you explore? (Optional)
NA

**Comment by strepanier03** [comment]
Hmmm, gonna think about this one and likely raise it, then circle back Tuesday next week when I'm back in office.

**Comment by strepanier03** [comment]
Update tomorrow.

**Comment by strepanier03** [comment]
All right, repro'd and raised in Slack [here](https://expensify.slack.com/archives/C03U7DCU4/p1705953039576839).

<img width="1440" alt="image" src="https://github.com/Expensify/App/assets/10925636/09eb2759-c6f7-4a0e-a727-bf109d1ba3c5">

**Comment by strepanier03** [comment]
All right, tied to Wave 5 and discussion in Slack produced good outcome. 

tl;dr: Rather than editing the message, we shouldn't be allowing an expense to be paid without a merchant.

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Enter merchant violation displayed even the expense is paid

### What is the root cause of that problem?
We don't have any logic to show/hide violation when the expense is paid or cancelled.

### What changes do you think we should make in order to solve the problem?
We can ignore showing any violation when the expense gets paid or cancelled here.
```canUseViolations && !isSettled && !isCancelled```
https://github.com/Expensify/App/blob/0bdbfbbc5ffa8a247f348abe777819eb44b8db1a/src/components/ReportActionItem/MoneyRequestView.js#L178

Also for stop showing the RBR indicator we need to add similar conditions in `ReportPreview`

**Comment by stitesExpensify** [comment]
Hey @c3024! I know you just got assigned as c+ yesterday but it looks like there are a couple of proposals if you wouldn't mind checking them out!

**Comment by c3024** [comment]
I'll check the proposals and update in a few hours.

**Comment by c3024** [comment]
@DylanDylann 's [proposal here](https://github.com/Expensify/App/issues/34415#issuecomment-1888774723) looks good to me.

🎀 👀 🎀 C+ Reviewed

**Comment by DylanDylann** [comment]
@c3024 The PR is ready for review https://github.com/Expensify/App/pull/35360

**Comment by strepanier03** [comment]
Looks like we're all set for tomorrow. @c3024, feel free to complete the checklist above and I'll do the reg test.

**Comment by c3024** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
 
* [x]  [@c3024] The PR that introduced the bug has been identified. Link to the PR: #26155 
* [x]  [@c3024] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/commit/03f6abee4691333e1da39634ce277f4627e04838#r138623871
* [x]  [@c3024] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: This is a specific case of settled request missed. A discussion was not started because it does not appear that this could have been identified earlier.
* [x]  [@c3024] Determine if we should create a regression test for this bug. Yes
* [x]  [@c3024] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal
1. Login with admin of a workspace on a device and member of the workspace on another device.
2. [Member] Go to workspace chat and create a scan request.
3. [Member] Manually enter the amount of the scanning request, while leaving Merchant empty.
4. [Admin] Pay the request.
5. [Admin] Navigate to the money request details view.
6. Verify that there is no error message nor RBR in merchant field.

👍 or 👎

**Comment by strepanier03** [comment]
Reg test done and contracts paid and closed. Thanks again @c3024 and @DylanDylann for helping get this one over the finish line!
